src/structures/WelcomeChannel.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const Base = require('./Base');
+const Emoji = require('./Emoji');
+
+/**
+ * Represents a channel link in a guild's welcome screen.
+ */
+class WelcomeChannel extends Base {
+  constructor(guild, data) {
+    super(guild.client);
+    /**
+     * The guild for this welcome channel
+     * @type {Guild}
+     */
+    this.guild = guild;
+
+    this._patch(data);
+  }
+
+  /**
+   * Builds the welcome channel with the provided data.
+   * @param {*} data The raw data of this welcome channel
+   * @private
+   */
+  _patch(data) {
+    if (!data) return;
+    /**
+     * The description of this welcome channel
+     * @type {string}
+     */
+    this.description = data.description;
+    this._emoji = {
+      name: data.emoji_name,
+      id: data.emoji_id,
+    };
+
+    /**
+     * The id of this welcome channel
+     * @type {Snowflake}
+     */
+    this.channelID = data.channel_id;
+  }
+
+  /**
+   * The channel of this welcome channel
+   * @type {?(TextChannel|NewsChannel)}
```

Comment:
```suggestion
   * @type {?TextChannel|NewsChannel}
```

Reply:
unsure, can't find any docs that suggest the former to be wrong

Reply:
https://esdoc.org/manual/tags.html#nullable-and-not-nullable

ES Doc seems to point towards the former, but I can't find anything specific for JSDoc

Reply:
It's not wrong, just seems unnecessary

Reply:
Found other instances in the docs where we apply the non-parens approach, commiting